src/Apps/Legion/Components/UserDefinedEntityForm/UserDefinedEntityForm.view.tsx

```diff
@@ -0,0 +1,415 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
```

Comment:
`.view.tsx`? Why not a new component name?

Reply:
Was trying something new. This is the renderer and the main component fetches the data.

Reply:
Naming sounds good, but this looks very complicated to just be a renderer. How much data manipulation could be passed down as a prop or be extracted to hooks?

Reply:
Well it takes care of knowing which type of form to show. In my opinion that is what the renderer should do. The parent doesn't need to know that, it just wants to know the output of the form and takes care of persisting that to state. Since both versions of the form result in the same data it seemed reasonable to keep that abstracted away from the parent.

Reply:
Is it the lookups to find existing entities that you're talking about? Otherwise, I think most of the logic here is just transforming data from the form back into a standard form to send to the parent. I did go back and forth though about whether to resolve the "existing" items in the renderer vs in the parent. I do think it'd probably have been "ideal" to do that in the parent, but that meant that the child then would have different props back based on the selections and I opted for the cleaner interface. You're right though, it would be nice if the form would just report out on what was selected or entered and not have to care about other data in the system.